pkg/cmd/roachtest/testselector/snowflake_query.sql

```diff
@@ -36,10 +41,15 @@ select
          failure_count > 0 or -- the test has failed at least once in the past "forPastDays" days
          first_run > dateadd(DAY, ?, current_date()) or -- recently added test - test has not run for more than "firstRunOn" days
          last_run < dateadd(DAY, ?, current_date()) or -- the test has not been run for last "lastRunOn" days
-         last_run is null -- the test is always ignored till now or have never been run
-         then 'yes' else 'no' end as selected,
+         -- as stated above, latest unsuccessful test should be selected. this will not consider any test
+         -- that has failed with non-infra error as the failure_count will not be zero.
+         recent_status != 'SUCCESS'
+  then 'yes' else 'no' end as selected,
   -- average duration - this is set to 0 if the test is never run (total_runs=0)
   case when total_runs > 0 then total_duration/total_runs else 0 end as avg_duration,
   total_runs,
+  -- indicates the tests failed and the last failure was due to infra flake
+  -- this information can be used later to ensure that we do not run this test on spot, so that this would succeed
+  case when failure_count=0 and recent_status='FAILURE' then 'yes' else 'no' end as last_failure_is_infra,
```

Comment:
Not sure I follow this predicate either. If test A failed two times in the past 7 days, once due to a bug and later (yesterday) due to preemption, won't `failure_count` be 1? In which case, `last_failure_is_infra` will be false, although the last failure _is_ a preemption. What am I missing?

Reply:
yes, you are right. Thanks for pointing out. I will actually need to add a condition to get the latest error message - this is doable.